From ea202c2aaa3c63ff1296fcc85e836efdb3af1cb2 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Thu, 17 Feb 2022 14:48:30 +0000
Subject: [PATCH] media: i2c: ov7251: Reinstate setting
 ov7251_global_init_setting

"media: i2c: Remove .s_power() from ov7251" removed the call that
sent ov7251_global_init_setting to the sensor. Send it when starting
streaming.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/i2c/ov7251.c | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/media/i2c/ov7251.c
+++ b/drivers/media/i2c/ov7251.c
@@ -1249,6 +1249,14 @@ static int ov7251_s_stream(struct v4l2_s
 		if (ret < 0)
 			return ret;
 
+		ret = ov7251_set_register_array(ov7251,
+					ov7251_global_init_setting,
+					ARRAY_SIZE(ov7251_global_init_setting));
+		if (ret < 0) {
+			dev_err(ov7251->dev, "could not set global_init_setting\n");
+			goto err_power_down;
+		}
+
 		ret = ov7251_pll_configure(ov7251);
 		if (ret) {
 			dev_err(ov7251->dev, "error configuring PLLs\n");
